### PR TITLE
[Sixgill-195] DVE feed

### PR DESCRIPTION
**Pull request by BenSterenson**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 5.0.0
- [x] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/Sixgill-Darkfeed/Integrations/Sixgill_DVE_Feed/Sixgill_DVE_Feed.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+from sixgill.sixgill_request_classes.sixgill_auth_request import SixgillAuthRequest
+from sixgill.sixgill_feed_client import SixgillFeedClient
+from sixgill.sixgill_constants import FeedStream
+from sixgill.sixgill_utils import is_indicator
+
+# Disable insecure warnings
+requests.packages.urllib3.disable_warnings()
+
+""" CONSTANTS """
+CHANNEL_CODE = "7698e8287dfde53dcd13082be750a85a"
```
Comment (reutshal) [inline Packs/Sixgill-Darkfeed/Integrations/Sixgill_DVE_Feed/Sixgill_DVE_Feed.py:20]: Are you OK with the fact that this channel code will be in a public github repo?

File: Packs/Sixgill-Darkfeed/Integrations/Sixgill_DVE_Feed/Sixgill_DVE_Feed.py [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+    Performs basic Auth request
+    """
+    response = SESSION.send(
+        request=SixgillAuthRequest(
+            demisto.params()["client_id"], demisto.params()["client_secret"], CHANNEL_CODE
+        ).prepare(),
+        verify=VERIFY,
+    )
+    if not response.ok:
+        raise Exception("Auth request failed - please verify client_id, and client_secret.")
```
Comment (reutshal) [inline Packs/Sixgill-Darkfeed/Integrations/Sixgill_DVE_Feed/Sixgill_DVE_Feed.py:70]: Shouldn't you return here the exception that was returned from the API?
What if there was another problem?

File: Packs/Sixgill-Darkfeed/Integrations/Sixgill_DVE_Feed/Sixgill_DVE_Feed.py [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+            "nvdlink": nvd_obj.get("link", ""),
+            "nvdlastmodifieddate": nvd_obj.get("modified", ""),
+            "nvdpublicationdate": nvd_obj.get("published", ""),
+            "cvss20score": nvd_obj.get("score_2_0", ""),
+            "cvss20severity": nvd_obj.get("severity_2_0", ""),
+            "nvdvectorv20": nvd_obj.get("vector_v2", ""),
+            "nvdvectorv31": nvd_obj.get("vector_v3", ""),
+        }
+    except Exception as err:
+        demisto.error(err)
```
Comment (reutshal) [inline Packs/Sixgill-Darkfeed/Integrations/Sixgill_DVE_Feed/Sixgill_DVE_Feed.py:111]: It is recommended to raise a DemistoException and let the main method handle it

File: Packs/Sixgill-Darkfeed/Integrations/Sixgill_DVE_Feed/Sixgill_DVE_Feed.py [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+        demisto.error(traceback.format_exc())
+    return fields
+
+
+def stix_to_indicator(stix_obj, tags: list = [], tlp_color: Optional[str] = None):
+    indicator: Dict[str, Any] = {}
+    try:
+        ext_obj = stix_obj.get("external_references", [])
+        ext_id = ""
+        if ext_obj[0]:
```
Comment (reutshal) [inline Packs/Sixgill-Darkfeed/Integrations/Sixgill_DVE_Feed/Sixgill_DVE_Feed.py:121]: This line will cause a 'list index out of range' error when ext_obj is an empty list.
I think it is better to do something like:
```suggestion
        if ext_obj and ext_obj[0]:
```

File: Packs/Sixgill-Darkfeed/Integrations/Sixgill_DVE_Feed/Sixgill_DVE_Feed.py [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+
+
+def create_fields(stix_obj, event_obj, nvd_obj, score_obj, ext_id):
+    fields = {}
+    try:
+        fields = {
+            "description": "",
+            "creationdate": stix_obj.get("created", ""),
+            "modified": stix_obj.get("modified", ""),
+            "externalid": ext_id,
```
Comment (reutshal) [inline Packs/Sixgill-Darkfeed/Integrations/Sixgill_DVE_Feed/Sixgill_DVE_Feed.py:89]: Looking at the code, it looks like there is a scenario that ext_id is an empty string. just making sure you are aware and tested this case.

File: Packs/Sixgill-Darkfeed/Integrations/Sixgill_DVE_Feed/Sixgill_DVE_Feed.py [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+        event_obj = stix_obj.get("x_sixgill_info", {}).get("event", {})
+        nvd_obj = stix_obj.get("x_sixgill_info", {}).get("nvd", {})
+        score_obj = stix_obj.get("x_sixgill_info", {}).get("score", {})
+        fields = create_fields(stix_obj, event_obj, nvd_obj, score_obj, ext_id)
+        fields = get_description(fields)
+        indicator["value"] = ext_id
+        indicator["type"] = "CVE"
+        indicator["rawJSON"] = {"value": ext_id, "type": "CVE"}
+        indicator["rawJSON"].update(stix_obj)
+        indicator["score"] = "3"
```
Comment (reutshal) [inline Packs/Sixgill-Darkfeed/Integrations/Sixgill_DVE_Feed/Sixgill_DVE_Feed.py:132]: Do all indicators have the same score?

File: Packs/Sixgill-Darkfeed/Integrations/Sixgill_DVE_Feed/Sixgill_DVE_Feed.py [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+        indicator["fields"] = fields
+        if tlp_color:
+            indicator["fields"]["trafficlightprotocol"] = str(tlp_color)
+        if tags:
+            indicator["fields"]["tags"] = ",".join(list(set(tags)))
+    except Exception as err:
```
Comment (reutshal) [inline Packs/Sixgill-Darkfeed/Integrations/Sixgill_DVE_Feed/Sixgill_DVE_Feed.py:138]: Same here, better to raise a DemistoException

File: Packs/Sixgill-Darkfeed/Integrations/Sixgill_DVE_Feed/Sixgill_DVE_Feed.yml [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+- display: Client Secret
+  hidden: false
+  name: client_secret
+  required: true
+  type: 4
+- defaultvalue: '1000'
+  hidden: false
+  name: maxIndicators
+  required: false
+  type: 0
```
Comment (reutshal) [inline Packs/Sixgill-Darkfeed/Integrations/Sixgill_DVE_Feed/Sixgill_DVE_Feed.yml:20]: Please add a display here like you did in your previous feed: 
image

File: Packs/Sixgill-Darkfeed/Integrations/Sixgill_DVE_Feed/Sixgill_DVE_Feed.yml [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+description: Leverage the power of Sixgill to supercharge Cortex XSOAR with real-time
+  Threat Intelligence indicators. Get CVE Feed straight into the XSOAR platform.
+display: Cybersixgill DVE Feed Threat Intelligence
+name: Sixgill_DVE_Feed
+script:
+  commands:
+  - arguments:
+    - default: true
+      defaultValue: '5'
+      description: 'No of records to display in War Room'
```
Comment (reutshal) [inline Packs/Sixgill-Darkfeed/Integrations/Sixgill_DVE_Feed/Sixgill_DVE_Feed.yml:102]:
```suggestion
      description: 'The number of records to display in War Room'
```

File: Packs/Sixgill-Darkfeed/Integrations/Sixgill_DVE_Feed/Sixgill_DVE_Feed.yml [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+  display: Fetch indicators
+  name: feed
+  required: false
+  type: 8
+- defaultvalue: '2'
+  display: Feed Fetch Interval
+  name: feedFetchInterval
+  required: false
+  type: 19
+- display: Bypass exclusion list
```
Comment (reutshal) [inline Packs/Sixgill-Darkfeed/Integrations/Sixgill_DVE_Feed/Sixgill_DVE_Feed.yml:36]:
```suggestion
- additionalinfo: When selected, the exclusion list is ignored for indicators from this
    feed. This means that if an indicator from this feed is on the exclusion list, the
    indicator might still be added to the system. 
display: Bypass exclusion list
```

Please add this additional info key in order to fix the validation error.
This parameter should look like this: 
```
- additionalinfo: When selected, the exclusion list is ignored for indicators from this
    feed. This means that if an indicator from this feed is on the exclusion list, the
    indicator might still be added to the system.
  display: Bypass exclusion list
  name: feedBypassExclusionList
  required: false
  type: 8```

**Review by reutshal — CHANGES_REQUESTED** [review]
Looks awesome!
Please see my notes.
- There are multiple flake8 errors in the test file + failing unit-tests that need to be fixed.
@BenSterenson Can you please share your availability for a short demo? Thanks

File: Packs/Sixgill-Darkfeed/Integrations/Sixgill_DVE_Feed/Sixgill_DVE_Feed.py (hunk 1 above)
Comment (BenSterenson) [reply Packs/Sixgill-Darkfeed/Integrations/Sixgill_DVE_Feed/Sixgill_DVE_Feed.py:20]: yes its not a secet :)

File: Packs/Sixgill-Darkfeed/Integrations/Sixgill_DVE_Feed/Sixgill_DVE_Feed.py (hunk 6 above)
Comment (BenSterenson) [reply Packs/Sixgill-Darkfeed/Integrations/Sixgill_DVE_Feed/Sixgill_DVE_Feed.py:132]: yes

**Review by BenSterenson — COMMENTED** [review]
thanks @reutshal, we will be working on the changes. ill reach out on slack to set a demo

File: Packs/Sixgill-Darkfeed/IndicatorFields/indicator_cvss20score.json [hunk 11]
```diff
@@ -0,0 +1,55 @@
+{
+	"incidentFields": [
+		{
```
Comment (reutshal) [inline Packs/Sixgill-Darkfeed/IndicatorFields/indicator_cvss20score.json:3]: Looks like something is off with this file, why are all the data inside the "incidentFields" key? 
It should be under the json root.
@EliorKedar Please confirm

Also, please add a "fromversion" key tp all indicator fields

**Review by reutshal — CHANGES_REQUESTED** [review]

File: Packs/Sixgill-Darkfeed/IndicatorFields/indicator_cvss20score.json (hunk 11 above)
Comment (EliorKedar) [reply Packs/Sixgill-Darkfeed/IndicatorFields/indicator_cvss20score.json:3]: Correct, should look as follow:
image

File: Packs/Sixgill-Darkfeed/ReleaseNotes/2_0_0.md [hunk 12]
```diff
@@ -2,3 +2,64 @@
 #### Integrations
 ##### New: Cybersixgill DVE Feed Threat Intelligence
 - Leverage the power of Sixgill to supercharge Cortex XSOAR with real-time Threat Intelligence indicators. Get CVE Feed straight into the XSOAR platform. (Available from Cortex XSOAR 5.5.0).
+
+#### Indicator Fields
+##### New: CVSS 2.0 Severity
+- New field
+
```
Comment (reutshal) [inline Packs/Sixgill-Darkfeed/ReleaseNotes/2_0_0.md:9]: @BenSterenson I would add here the description of each field.

Comment (BenSterenson) [reply Packs/Sixgill-Darkfeed/ReleaseNotes/2_0_0.md:9]: @reutshal done

**Comment by sangsoojun88** [comment]
everything looks good on my end

**Review by reutshal — APPROVED** [review]

**Pull request merged**